editor/edit-post/modes/visual-editor/index.js

```diff
@@ -61,6 +62,7 @@ class VisualEditor extends Component {
 						showContextualToolbar={ ! this.props.hasFixedToolbar }
 					/>
 					<DefaultBlockAppender />
+					<Slot name="BlockList.Siblings" />
```

Comment:
Could you elaborate on the relationship between this slot and the `BlockList` component? I'm trying to determine how this might impact #3745, where nested blocks can render their own `BlockList` elements. Would we want those nested `BlockList`s to render their own slots? Or is this more accurately described by its relation relative the visual editor root?

Reply:
The latter, yes:

> more accurately described by its relation relative the visual editor root

I merged this into your work on https://github.com/WordPress/gutenberg/pull/3745 and it does fine as-is. However, given your awesome changes in that PR, I could move this slot up one level in the DOM and call it `WritingFlow.Siblings` if you prefer.

Reply:
> Would we want those nested BlockLists to render their own slots?

That won't be necessary. The link modal just needs to be in a stacking context that's outside of any BlockList. That way it can have a z-index that lays it on top of all blocks in the list, nested or otherwise. So a single Slot outside of the parent BlockList component will do.

Reply:
>I could move this slot up one level in the DOM and call it `WritingFlow.Siblings` if you prefer.

Yes, especially since it seems we wouldn't want the content rendered by this slot to be considered in the writing flow behaviors managed by `WritingFlow`, this seems sensible. Alternatively maybe still better to clarify this as relative the visual editor, e.g. `VisualEditor.LastChild` (not sure what naming convention we'd want to use here for end/beginning of content)

Reply:
@aduth `VisualEditor.LastChild` seems fine to me. I rebased, resolved conflicts, and made the adjustment we discussed. Link modal now in the `VisualEditor.LastChild` slot.